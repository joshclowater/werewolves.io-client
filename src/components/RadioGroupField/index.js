import React from 'react';
import styled from 'styled-components';
import { Field } from 'redux-form';
import FieldWrapper from '../FieldWrapper';

const Radio = styled.div`
  display: block;
  margin-bottom: 1vh;
`;

export const RenderField = ({
  options,
  input,
  label,
  meta: { touched, error, warning, submitting },
  ...rest
}) => (
  <FieldWrapper
    inputName={input.name}
    touched={touched}
    error={error}
    warning={warning}
  >
    {options.map(option => (
      <Radio key={option}>
        <input
          type="radio"
          {...input}
          {...rest}
          id={`${option}Option`}
          value={option}
          checked={option === input.value}
          disabled={!!submitting}
        />
        <label id={`${option}Label`} htmlFor={`${option}Option`}>
          {option}
        </label>
      </Radio>
    ))}
  </FieldWrapper>
);

export default function RadioGroupField(props) {
  return <Field {...props} component={RenderField} />;
}
