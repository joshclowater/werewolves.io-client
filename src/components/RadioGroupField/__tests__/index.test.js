import React from 'react';
import { shallow } from 'enzyme';
import RadioGroupField, { RenderField } from '../';

it('RadioGroupField renders correctly', () => {
  expect(shallow(<RadioGroupField anyProp="mock prop" />)).toMatchSnapshot();
});

it('RenderField renders correctly', () => {
  expect(
    shallow(
      <RenderField
        input={{ name: 'mockName', value: 'mock value' }}
        options={['mock value', 'mock value 2']}
        meta={{
          touched: true,
          error: 'mock error',
          warning: 'mock warning',
          submitting: true
        }}
      />
    )
  ).toMatchSnapshot();
});
