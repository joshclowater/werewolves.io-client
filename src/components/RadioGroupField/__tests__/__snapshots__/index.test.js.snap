// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroupField renders correctly 1`] = `
<Field
  anyProp="mock prop"
  component={[Function]}
/>
`;

exports[`RenderField renders correctly 1`] = `
<FieldWrapper
  error="mock error"
  inputName="mockName"
  touched={true}
  warning="mock warning"
>
  <styled.div
    key="mock value"
  >
    <input
      checked={true}
      disabled={true}
      id="mock valueOption"
      name="mockName"
      type="radio"
      value="mock value"
    />
    <label
      htmlFor="mock valueOption"
      id="mock valueLabel"
    >
      mock value
    </label>
  </styled.div>
  <styled.div
    key="mock value 2"
  >
    <input
      checked={false}
      disabled={true}
      id="mock value 2Option"
      name="mockName"
      type="radio"
      value="mock value 2"
    />
    <label
      htmlFor="mock value 2Option"
      id="mock value 2Label"
    >
      mock value 2
    </label>
  </styled.div>
</FieldWrapper>
`;
